proj/src/main/webapp/chat.js

```diff
@@ -452,6 +457,11 @@ function pushChatMessage() {
             'timestamp': message.timestamp
         });
         notifications.sendNotificationForChat(message)
+
+        const unreadClasses = document.querySelectorAll('.unread');
+        unreadClasses.forEach(unreadMarker => {
+            unreadMarker.innerText = '';
+        });
```

Comment:
Do you also need to update the lastVisted value in the database here?

Reply:
It doesn't really make a difference. Either way, any new messages will be more recent than the last time the user has viewed the page, and they'll receive notifications when we want them to. I was thinking we might as well not update when we don't have to and save some client-side compute time, no?